rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf

```diff
@@ -560,6 +560,35 @@ SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|ARGS_NAME
     setvar:'tx.inbound_anomaly_score_pl1=+%{tx.critical_anomaly_score}'"
 
 
+# This rule catches on Scientific Notation bypass payloads in MySQL
+# Reference: https://github.com/swisskyrepo/PayloadsAllTheThings/blob/master/SQL%20Injection/MySQL%20Injection.md#scientific-notation
+#
+# Regular expression generated from regex-assembly/942560.ra.
+# To update the regular expression run the following shell script
+# (consult https://coreruleset.org/docs/development/regex_assembly/ for details):
+#   crs-toolchain regex update 942560
+#
+SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|ARGS_NAMES|ARGS|XML:/* "@rx (?i)1\.e[\(-\),]" \
```

Comment:
@theseion Why do we get the `-` in this regex based on the source file above? Any why at this position?

Reply:
That's Go's regular expression library. They prefer to write ranges in character classes, which sometimes leads to this output. It's semantically correct.

Reply:
Fascinating. So this is a range, but the range only has two elements?

Reply:
Correct.

Reply:
Thanks